### PR TITLE
Split single file

### PR DESCRIPTION
**Pull request by sdsykes**
The single file approach to FastImage has got rather unwieldy. Time to split it up, this is a first pass at that. 

No functionality changes.

### CODE REVIEW

File: lib/fastimage/fastimage_parsing/exif.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+class Exif # :nodoc:
```
Comment (sdsykes) [inline lib/fastimage/fastimage_parsing/exif.rb:1]: Module missing here

File: lib/fastimage.rb [hunk 2]
```diff
@@ -62,1005 +62,28 @@
 require 'base64'
 require 'uri'
 
+require 'fastimage/fastimage_parsing/stream_util'
```
Comment (SamSaffron) [inline lib/fastimage.rb:65]: I tend to either autoload or require_relative

require certainly works but it has to walk the loadpath.

File: lib/fastimage/fastimage.rb [hunk 3]
```diff
@@ -0,0 +1,405 @@
+class FastImage
+  include FastImageParsing
+
+  VERSION = "2.2.3"
```
Comment (SamSaffron) [inline lib/fastimage/fastimage.rb:4]: oh I missed this one when I just released. I think we should just move it to a single file, then you can source it from the gemspec

eg: https://github.com/ruby/rake/blob/master/rake.gemspec#L2-L3

**Comment by SamSaffron** [comment]
This is a great change, should make it easier to maintain long term!

**Review by Fazsarah — APPROVED** [review]

**Review by Fazsarah — APPROVED** [review]

**Review by Fazsarah — APPROVED** [review]

**Review by Fazsarah — APPROVED** [review]

**Review by Fazsarah — APPROVED** [review]

**Review by Fazsarah — APPROVED** [review]
Great

**Review by Fazsarah — APPROVED** [review]

**Review by Fazsarah — APPROVED** [review]

**Comment by sdsykes** [comment]
Superseded by a newer PR

**Pull request closed without merging**